creating_citation_counts_tsv/scripts/summarize_gene_citations_all_species.py

```diff
@@ -2,270 +2,215 @@
 Creating TSVs containing the 10 most-cited genes per species and their gene information.
 """
 import os
-import sys
-import json
-import pyspark
-import requests
-from operator import add
 from datetime import date, timedelta, datetime
 import csv
-from collections import Counter
 
 
-def get_recent_gene2pubmed(species):
+def get_recent_gene_citation_count(species_gene2pubmed_tsv, recent_citations_ssv):
     """ 
-    Getting a list of the gene ID, gene publication citation ID, and count of how frequently the gene is cited over the past 5 months for each gene.
+    Input:
+        species_gene2pubmed_tsv: tsv of genes and associated publication citation ID for a species
+        recent_citations_ssv: ssv of all recent citations
+    Output:
+        Count of recent publication citations for each gene ID.
     """
 
-    # Getting a list of the gene ID and gene publication citation ID pairs and saving it to the gene2pubmed variable 
-    gene2pubmed = (sc.textFile(f"creating_citation_counts_tsv/data/{species}/gene2pubmed")
```

Comment:
for future reference:

names of variables did not match up to the tuples -- things were backwards or included more than indicated by the variable name.

variables disambiguated:

```
#gene2pubmed = (pmid, gid)
#year_pmid = (pmid, year)
#gene2pubmed_to_year = (gid, pmid)
#gene_counts = (geneid, (pmid, count))
#gene_info  = (Symbol.upper(), (GeneID, chromosome, description, count, tax_id))
#ref_gene = (gene_symbol, (chromosome, start_coord, coord_length, "#73af42", description, count))

# how I got these values, aka "show your work" (this is not pretty as it is a bit stream-of-conscience):

# gene_counts = gene2pubmed.join(year_pmid)
#                 .map(lambda x: (x[1][0], 1)) # geneid
#                 .reduceByKey(add)
#                 .join(gene2pubmed_to_year) # (geneid, (count, pmid))
#                 .map(lambda x: ( str(x[0]), (x[1][1] ,x[1][0]))) # (geneid, (pmid, count))

# gene_info file excerpt:
# #tax_id   GeneID  Symbol  LocusTag        Synonyms        dbXrefs chromosome      map_location
#  description     type_of_gene    Symbol_from_nomenclature_authority      Full_name_from_nomenclature_authority   Nomenclature_status     Other_designations      Modification_date       Feature_type
#7       5692769 NEWENTRY        -       -       -       -       -       Record to support submission of GeneRIFs for a gene not in Gene (Azotirhizobium caulinodans.  Use when strain, subtype, isolate, etc. is unspecified, or when different from all specified ones in Gene.).      other   -       -       -       -       20210407        -
#159     63961479        recO    GQX59_RS00690   GQX59_00690     -       -       -       DNA repair protein RecO protein-coding  -       -       -       DNA repair protein RecO 20210611
#        -

# orig gene_info code
# ( str(x[1]), ( str(x[0]), str(x[2]), str(x[6]), str(x[8] ) )
# ( GeneID, ( tax_id, Symbol, chromosome, description ) )
# GeneID = str(x[1]
# tax_id = (str(x[0])
# Symbol = str(x[2])
# chromosome = str(x[6])
# description = str(x[8]))

    # gene_info = (sc.textFile(species_gene_info_tsv)
    #                 .filter(lambda x: x[0] != '#')
    #                 .map(lambda x: x.split('\t'))
    #                 .map(lambda x: (str(x[1]), (str(x[0]), str(x[2]), str(x[6]), str(x[8]))))) # ( GeneID, ( tax_id, Symbol, chromosome, description ) )
    #                   .join(gene_citation_counts) # ( GeneID, ( (tax_id, Symbol, chromosome, description), (pmid, count) ) )
    #                   .map(lambda x: (x[1][0][1].upper(), (x[0], x[1][0][2], x[1][0][3], x[1][1][1], x[1][0][0])))
    #(x[1][0][1].upper(), (x[0], x[1][0][2], x[1][0][3], x[1][1][1], x[1][0][0]))
    #(Symbol.upper(), (GeneID, chromosome, description, count, tax_id))
    # Symbol = x[1][0][1] #.upper()
    # GeneID = x[0] #(above info from gene_info)
    # chromosome = x[1][0][2]
    # description = x[1][0][3]
    # count = x[1][1][1]
    # tax_id = x[1][0][0]



            refGene = (sc.textFile(f"creating_citation_counts_tsv/data/{species}/{refGene_file}")
                    .filter(lambda x: x[0] != '#')
                    .map(lambda x: x.split('\t'))
                    .filter(lambda x: 'transcript' in x[2]) # type (transcript, exon, 3utr, cds ...)
                    .map(lambda x: ( str(x[8]) # example: """gene_id "TMEM181"; transcript_id "NR_164859"; exon_number "16"; exon_id "NR_164859.16"; gene_name "TMEM181";"""
                            .split(";")[0]
                            .replace("gene_id ", "")
                            .replace('"', '')
                            .upper()
                        , (int(x[3]), int(abs(int(x[3])-int(x[4]))))  )) # start, start, end
                        # ( gene_symbol, (start_coord, coord_length) )
                    .join(gene_symbol__gene_info__dict) # ( gene_symbol, ((start_coord, coord_length), (GeneID, chromosome, description, count, tax_id)) )
                    .map(lambda x: (x[0], (x[1][1][1], x[1][0][0], x[1][0][1], "#73af42", x[1][1][2], x[1][1][3])))
                    #.map(lambda x: (gene_symbol, (chromosome, start_coord, coord_length, "#73af42", description, count)))
                    .reduceByKey(lambda a, b: a)
                    .map(lambda x: (x[0], x[1][0], x[1][1], x[1][2], "#73af42", x[1][4], x[1][5])))
                    #.map(lambda x: (gene_symbol, (chromosome, start_coord, coord_length, "#73af42", description, count)))


```